### PR TITLE
Create module-info.java

### PR DESCRIPTION
**Pull request by azabbasi**
For more information:
https://www.jrebel.com/blog/java-9-modules-cheat-sheet

### CODE REVIEW

**Comment by timtay-microsoft** [comment]
Is there some way to generate this file? Or is this hand-written?

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    requires com.azure.http.netty;
+
+    opens com.azure.digitaltwins.core to come.azure.core, com.azure.http.netty;
+    opens com.azure.digitaltwins.core.models to com.fasterxml.jackson.annotation, come.azure.core;
+    opens com.azure.digitaltwins.core.util to com.fasterxml.jackson.annotation, come.azure.core;
+    opens com.azure.digitaltwins.core.serialization to com.fasterxml.jackson.annotation, come.azure.core;
+    opens com.azure.digitaltwins.core.implementation to com.fasterxml.jackson.annotation, come.azure.core;
+    opens com.azure.digitaltwins.core.implementation.models to com.fasterxml.jackson.annotation, com.azure.core;
+    opens com.azure.digitaltwins.core.implementation.serializer to com.fasterxml.jackson.annotation, com.azure.core;
+    opens com.azure.digitaltwins.core.implementation.converters to com.fasterxml.jackson.annotation, com.azure.core;
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:16]: how did you come up with this list? is there a script that lists them?

Comment (azabbasi) [reply sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:16]: Unfortunately there is none ! I used the cheat sheet and my own judgement along with help from @alzimmermsft

**Comment by azabbasi** [comment]
> Is there some way to generate this file? Or is this hand-written?

Hand written !!

**Comment by azabbasi** [comment]
I am going to use the build errors to fix all the missing ones.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module come.azure.digitaltwins.core {
```
Comment (srnagar) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:4]:
```suggestion
module com.azure.digitaltwins.core {
```

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    requires transitive com.azure.core;
+    requires com.fasterxml.jackson.annotation;
+    requires com.azure.http.netty;
```
Comment (srnagar) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:7]: Jackson should be available transitively through azure core and netty should not be included in this list. Http clients are pluggable and netty can be replaced by another http client.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    opens com.azure.digitaltwins.core to come.azure.core, com.azure.http.netty;
```
Comment (srnagar) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:9]: I am not sure if we should open anything up specifically for netty.

File: sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    opens com.azure.digitaltwins.core.models to com.fasterxml.jackson.annotation, come.azure.core;
+    opens com.azure.digitaltwins.core.util to com.fasterxml.jackson.annotation, come.azure.core;
```
Comment (srnagar) [inline sdk/digitaltwins/azure-digitaltwins-core/src/main/java/module-info.java:11]: Why do we need to open this for core?

**Comment by alzimmermsft** [comment]
For the Java 11 test failures you'll want to add a plugin configuration similar to this to the `pom.xml`.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/appconfiguration/azure-data-appconfiguration/pom.xml#L98

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**